### ISSUE TITLE
C++ snippets regression

### ISSUE THREAD

**Issue by Alexey-T** [OlehL/cuda_snippets#31]
Snippet Std.C/struct.synw-snippet
```
name=struct
id=struct
lex=C,C++,Objective-C
text=
struct {${1:name}}
{
	${0}
};
```

plugin cannot insert it. "Wrong snippet".
I added debug lines to parse_tabstops:

```
                if is_tabstop(t[0]):
                    print('#is_tabstop', t[0])
                    _tag = int(t[1])
                    m = marker(
                        x=t.start(0) + shift + (x0 if y == 0 else 0),
                        y=y+y0,
                        tag=_tag + basetag
                    )
                    if _tag == 0:
                        zero_markers.append(m)
                    else:
                        markers.append(m)
                    shift -= len(t[0])

                elif is_placeholder_head(t[0]):
                    print('#is_placeh_head', t[0])
                    p = Placeholder(
                        x0=t.start(0),
                        x1=t.end(0),
                        y=y,
                        shift=shift,
                        tag=int(t[2])
                    )
                    buf.append(p)
                    shift -= t.end(0) - t.start(0)

                elif is_placeholder_tail(t[0]):
                    print('#is_placeh_tail', t[0])
                    if not buf:
```

and see that plugin prints-

```
Init: cuda_snippets
#is_placeh_head ${1:
#is_placeh_tail }
#is_placeh_tail }
Wrong snippet. name: struct, sn: ['struct {${1:name}}', '{', '    ${0}', '};'], markers: None
```
It fails when finds second `}`. it is NOT from tabstop, it is C++ bracket.
```
struct {${1:name}}
```
fails here
```
                elif is_placeholder_tail(t[0]):
                    print('#is_placeh_tail', t[0])
                    if not buf:
                        return None, None, None
```
How to fix?
@OlehL

**Comment by OlehL** [comment]
I will try to fix it at evening.

**Comment by OlehL** [comment]
fixed in 11b41a7b6eba330e35e3b3d5ca877624900cb6d4
pls test

**Comment by Alexey-T** [comment]
Now new error for snippet "struct" from C++ snippets:
```
Python 3.8
Init: cuda_snippets
Startup: 350ms, plugins: 40ms (snippets 42ms)
Init: cuda_hilite_occurrences
Traceback (most recent call last):
  File "/home/user/.config/cudatext/py/cuda_snippets/__init__.py", line 51, in on_key
    items = [i for i in self.lex_snippets if name in i.id]  # leave snips for name
  File "/home/user/.config/cudatext/py/cuda_snippets/__init__.py", line 39, in lex_snippets
    return self.th.snippets.get(lexer, []) + self.th.glob
TypeError: can only concatenate list (not "dict") to list
```

**Comment by OlehL** [comment]
fix in 975f39f
